File: src/main/java/com/lowtuna/jsonblob/view/AboutView.java
package com.lowtuna.jsonblob.view;

import com.lowtuna.dropwizard.extras.config.GoogleAnalyticsConfig;
import io.dropwizard.util.Duration;
import lombok.Data;

import java.util.Set;

@Data
public class AboutView extends JsonBlobView {
    private final Duration blobAccessTtl;
    private final boolean deletionEnabled;

    public AboutView(String gaWebPropertyID, String pageName, Set<GoogleAnalyticsConfig.CustomTrackingCode> customTrackingCodes, Duration blobAccessTtl, boolean deletionEnabled) {
        super("/about.hbs", gaWebPropertyID, pageName, customTrackingCodes);
        this.blobAccessTtl = blobAccessTtl;
        this.deletionEnabled = deletionEnabled;
    }
}

File: src/main/kotlin/jsonblob/api/http/view/AboutView.kt
package jsonblob.api.http.view

import io.micronaut.core.annotation.Introspected
import jsonblob.config.GoogleAnalyticsConfig

@Introspected
class AboutView(
    gaWebPropertyID: String,
    pageName: String,
    gaCustomTrackingCodes: Set<GoogleAnalyticsConfig.CustomTrackingCode>,
    val deleteAfterDays: Long
) : JsonBlobView(gaWebPropertyID, pageName, gaCustomTrackingCodes)